### PR TITLE
Ignore null / invalid spancontext and handle null attributes for links.

### LINKED ISSUE #2955: Adding an invalid span as a link
**Describe the bug**
What should happen when you add an invalid span (traceId=0, spanId=0) as a link?
Currently such a link is added to the span and even exported (I haven't tested it, just read the code: jaeger & otlp exporters do not filter links at all, nor does SDK). That's quite different to how we treat parent contexts - where we check if the context contains a valid span (in `SdkSpanBuilder#startSpan()`).
Is the current behavior correct?

### PR DESCRIPTION
**Pull request by anuraaga**
Fixes #2955

Spec PR :https://github.com/open-telemetry/opentelemetry-specification/pull/1492

### CODE REVIEW

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java [hunk 1]
```diff
@@ -83,12 +83,24 @@ public SpanBuilder setSpanKind(SpanKind spanKind) {
...
     addLink(LinkData.create(spanContext));
     return this;
   }
 
   @Override
   public SpanBuilder addLink(SpanContext spanContext, Attributes attributes) {
+    if (spanContext == null || !spanContext.isValid()) {
+      return this;
+    }
+    if (!spanContext.isValid()) {
```
Comment (jkwatson) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java:98]: this is redundant with the check above, correct?

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java [hunk 2]
```diff
@@ -83,12 +83,24 @@ public SpanBuilder setSpanKind(SpanKind spanKind) {
...
   public SpanBuilder addLink(SpanContext spanContext) {
+    if (spanContext == null || !spanContext.isValid()) {
+      return this;
+    }
     addLink(LinkData.create(spanContext));
     return this;
   }
 
   @Override
   public SpanBuilder addLink(SpanContext spanContext, Attributes attributes) {
```
Comment (bogdandrutu) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java:94]: Should we also document in the API, same as you din in the specs:

"Implementation MAY ignore ...."

Comment (anuraaga) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java:94]: Sure

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java (hunk 1 above)
Comment (anuraaga) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java]: Oops

**Review by bogdandrutu — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]

**Pull request merged**